File: go/classify.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// ClassifyService — V2 content classification.
type ClassifyService struct {
	t *internal.Transport
}

type ClassifyV2Request struct {
	Text string `json:"text"`
}

type ClassifyV2BatchRequest struct {
	Texts []string `json:"texts"`
}

func (s *ClassifyService) Classify(ctx context.Context, text string) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v2/classify", ClassifyV2Request{Text: text}, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *ClassifyService) Batch(ctx context.Context, texts []string) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v2/classify/batch", ClassifyV2BatchRequest{Texts: texts}, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *ClassifyService) Categories(ctx context.Context) (any, error) {
	var resp any
	if err := s.t.Do(ctx, "GET", "/v2/classify/categories", nil, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

File: python/src/aegis/resources/classify.py
"""Classify resource — V2 content classification endpoints."""

from __future__ import annotations

from typing import Any, Dict, List

from aegis.resources._base import AsyncResource, SyncResource


def _to_text(item: Any) -> str:
    if isinstance(item, str):
        return item
    if isinstance(item, dict):
        return item.get("text") or item.get("content") or ""
    return getattr(item, "text", None) or getattr(item, "content", None) or ""


class SyncClassify(SyncResource):

    def classify(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v2/classify", json={"text": content, **kwargs})

    def batch(self, requests: List[Any]) -> Dict[str, Any]:
        return self._post(
            "/v2/classify/batch",
            json={"texts": [_to_text(r) for r in requests]},
        )

    def categories(self) -> List[Dict[str, Any]]:
        data = self._get("/v2/classify/categories")
        return data if isinstance(data, list) else data.get("categories", [])


class AsyncClassify(AsyncResource):

    async def classify(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v2/classify", json={"text": content, **kwargs})

    async def batch(self, requests: List[Any]) -> Dict[str, Any]:
        return await self._post(
            "/v2/classify/batch",
            json={"texts": [_to_text(r) for r in requests]},
        )

    async def categories(self) -> List[Dict[str, Any]]:
        data = await self._get("/v2/classify/categories")
        return data if isinstance(data, list) else data.get("categories", [])
